### DISCUSSION TITLE
Can translation string parameters have fallbacks?

### DISCUSSION THREAD

**Discussion by leo-mangold** [ivanhofer/typesafe-i18n#705] [General]
I have this in my translation:
``` js
{
  preview: "Preview {title}"
}
```
This is great because then i can do something like `$LL.preview({title: "Session"})`.
But I was wondering if there is a possibility to add a fallback to the parameters, so if I don't have a title yet I could show "Preview Document" or whatever, so my translation object would look something like this:
```js
{
  preview: "Preview {title ?? 'Document'}"
}
```

I know I can do `$LL.preview({title: myCoolTitle ?? "Document"})`, but that would kinda defeat the use of translations.
Are there any workarounds for this?

**Comment by ivanhofer** [comment]
No, there are no fallbacks for parameters.
You can edit your example above like this to also have translated fallbacks:
```ts
$LL.preview({ title: myCoolTitle ?? $LL.titleFallback() })
```
Hope this helps :)

**Comment by leo-mangold** [comment]
Okey thats a bummer but thanks anyway
